packages/indexer-common/src/indexer-management/allocations.ts

```diff
@@ -482,6 +485,33 @@ export class AllocationManager {
     }
   }
 
+  async waitForBlockNumber(logger: Logger, blockNumber: number): Promise<void> {
+    await this.waitForBlockNumberOnEventual(
+      logger,
+      blockNumber,
+      this.network.networkSubgraph.deployment!.status.map(
+        (status) => status.latestBlock?.number ?? 0,
+      ),
+    )
+  }
+
+  async waitForBlockNumberOnEventual(
+    logger: Logger,
+    number: number,
+    sequentialNumberStream: Eventual<number>,
+  ): Promise<void> {
+    for await (const latestSequentialNumber of sequentialNumberStream.values()) {
+      if (latestSequentialNumber >= number) {
+        return
```

Comment:
Are there possible weird edge cases that could lead to this loop running indefinitely? 

If so maybe there is an actionable error message we could present? Or an exit strategy such as a timeout?